apps/balancer-tools/src/app/apr/(utils)/tokenYield.ts

```diff
@@ -81,37 +83,53 @@ function getAPRFromRate(
     // eslint-disable-next-line no-console
     console.error("Negative APR");
   }
-
   return APR < 0 ? 0 : APR;
 }
 
 async function getPoolTokensRateProviders(chain: string, poolId: Address) {
   const data = await pools.gql(String(chain)).PoolRateProviders({ poolId });
 
-  if (!data.pool?.priceRateProviders?.length)
+  if (!data.pool?.priceRateProviders?.length) {
```

Comment:
would we eventually use the manual approach to override bogus return values as well ? If so, we'd probably need to remove this if statement, right?